omeroweb/webadmin/urls.py

```
@@ -81,7 +81,7 @@
         name="waloaddrivespace_user",
     ),
     url(
-        r"^change_avatar/(?P<eid>[0-9]+)/(?:(?P<action>[a-z]+)/)?$",
+        r"^change_avatar/(?:(?P<action>[a-z]+)/)?$",
```

The biggest downside of this approach is that it will break all calls of type `https://myserver/webadmin/change_avatar/xxx/{upload,deletephoto}/` in favor of `https://myserver/webadmin/change_avatar/{upload,deletephoto}/`.

From my understanding of the associated code, the passed ID was silently ignored anyways and the current user ID used under the hood. Was the ability to set/delete other's avatar ever functional?

Trying to make the minimal assumptions about consumers, is it possible to update this to keep supporting both forms of URLs but use the same view?

---

This is a breaking change, so we should try to handle both url formats if possible

---

I can certainly allow `change_avatar/xxx/{upload,deletephoto}/` but for how long do we need to support it and how would we mark it as deprecated?
I don't think there was ever the ability to upload or delete another user's photo and I think it very unlikely that URL would be used by any external application (delete was even unusable in webclient till just now (#274).
So the sooner we can remove it the better.

---

Was the ID previously always ignored even if for another user?

---

Yes. As you can see in `master` (and not changed in this PR) `manageavatar()` view function makes no use of any `eid`: https://github.com/ome/omero-web/blob/688acfe1cae17d98b61309e9da18f775ce90fd91/omeroweb/webadmin/views.py#L1089
Same in [v5.9.1](https://github.com/ome/omero-web/blob/v5.9.1/omeroweb/webadmin/views.py#L1088) etc.

---

:+1: I was wondering if that means we should disallow now eid != current user.